src/librustc_data_structures/bitvec.rs

```diff
@@ -17,14 +17,15 @@ pub struct BitVector {
 }
 
 impl BitVector {
+    /// Create a new bitvector of at least num_bits length.
     pub fn new(num_bits: usize) -> BitVector {
         let num_words = u64s(num_bits);
         BitVector { data: vec![0; num_words] }
     }
 
     pub fn contains(&self, bit: usize) -> bool {
         let (word, mask) = word_mask(bit);
-        (self.data[word] & mask) != 0
+        (self.data.get(word).cloned().unwrap_or(0) & mask) != 0
```

Comment:
Oh, I see now.  Yes it should be consistent at least :)

> On Aug 11, 2016, at 4:14 PM, Simonas Kazlauskas <notifications@github.com> wrote:
> 
> In src/librustc_data_structures/bitvec.rs <https://github.com/rust-lang/rust/pull/35608#discussion_r74520803>:
> 
> >      pub fn new(num_bits: usize) -> BitVector {
> >          let num_words = u64s(num_bits);
> >          BitVector { data: vec![0; num_words] }
> >      }
> >  
> >      pub fn contains(&self, bit: usize) -> bool {
> >          let (word, mask) = word_mask(bit);
> > -        (self.data[word] & mask) != 0
> > +        (self.data.get(word).cloned().unwrap_or(0) & mask) != 0
> It seems to me that asking for an index that doesn't exist should panic and that this might mask bugs in other code.
> 
> It had such a behaviour for sizes which are not 0 mod 64 up to next 0 mod 64 bit already. This makes the behaviour consistent for all sizes.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/rust-lang/rust/pull/35608/files/0962575160c619b29df676c0fbb0cee3f804caf7#r74520803>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAc1nWMtSQhjePDC7v--c3TRfwBd_6YFks5qe6zdgaJpZM4Jinmf>.
>